62.CAD/fragment_shader.hlsl

```diff
@@ -11,10 +12,72 @@ void beginInvocationInterlockEXT();
 void endInvocationInterlockEXT();
 #endif
 
+//#define NBL_DRAW_ARC_LENGTH
+
 // TODO[Lucas]: have a function for quadratic equation solving
 // Write a general one, and maybe another one that uses precomputed values, and move these to somewhere nice in our builtin hlsl shaders if we don't have one
 // See: https://github.com/erich666/GraphicsGems/blob/master/gems/Roots3And4.c
 
+template<typename float_t, typename ArcLenCalculator>

```

Comment:
1. instead of `typename ArcLenCalculator` get a `typename CurveType`
and use `CurveType::ArcLenCalculator` instead.

that requires every CurveType using this clipper to have a `ArcLenCalculator` defined.

+ inside the Quadratic struct do `typedef AnalyticArcLenCalculator ArcLenCalculator`

2. instead of `quadratic` mention `curve` this struct should be general enough so that we can use it for a line.

Reply:
> inside the Quadratic struct do `typedef AnalyticArcLenCalculator ArcLenCalculator`

can't do nested typedefs in HLSL, I would rename `AnalyticArcLenCalculator` to `ArcLenCalculator`

Reply:
> > inside the Quadratic struct do `typedef AnalyticArcLenCalculator ArcLenCalculator`
> 
> can't do nested typedefs in HLSL, I would rename `AnalyticArcLenCalculator` to `ArcLenCalculator`

DXC is stupid, here's the thing:
1. you can use `typedef` but only in global space
2. you can use `using` but only nested inside a `struct` 

enjoy

Reply:
done